Xlib/ext/nvcontrol.py

```diff
@@ -0,0 +1,5358 @@
+    if not reply._data.get('flags'):
+        return None
+    return int(reply._data.get('min')), int(reply._data.get('max'))
+
+
+def query_binary_data(self, target, display_mask, attr):
+    """return binary data"""
+    reply = NVCtrlQueryBinaryDataReplyRequest(display=self.display,
+                                              opcode=self.display.get_extension_major(extname),
+                                              target_id=target.id(),
+                                              target_type=target.type(),
+                                              display_mask=display_mask,
+                                              attr=attr)
+    if not reply._data.get('flags'):
+        return None
+    return reply._data.get('data')
+
+
+def get_coolers_used_by_gpu(self, target):
+    reply = NVCtrlQueryListCard32ReplyRequest(display=self.display,
+                                              opcode=self.display.get_extension_major(extname),
+                                              target_id=target.id(),
+                                              target_type=target.type(),
+                                              display_mask=0,
+                                              attr=NV_CTRL_BINARY_DATA_COOLERS_USED_BY_GPU)
+    if not reply._data.get('flags'):
+        return None
+    return reply._data.get('list')
+
+
+def get_gpu_count(self):
+    """Return the number of GPU's present in the system."""
+    return int(query_target_count(self, Gpu()))
+
+
+def get_name(self, target):
+    """the GPU product name on which the specified X screen is running"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_PRODUCT_NAME)
+
+
+def get_driver_version(self, target):
+    """the NVIDIA (kernel level) driver version for the specified screen or GPU"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_NVIDIA_DRIVER_VERSION)
+
+
+def get_vbios_version(self, target):
+    """the version of the VBIOS for the specified screen or GPU"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_VBIOS_VERSION)
+
+
+def get_gpu_uuid(self, target):
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_GPU_UUID)
+
+
+def get_gpu_utilization(self, target):
+    string = query_string_attribute(self, target, 0, NV_CTRL_STRING_GPU_UTILIZATION)
+    result = {}
+    if string is not None and string != '':
+        for line in string.split(','):
+            key_value = line.split('=')
+            result[key_value[0].strip()] = int(key_value[1]) if key_value[1].isdigit else key_value[1]
+    return result
+
+
+def get_performance_modes(self, target):
+    string = query_string_attribute(self, target, 0, NV_CTRL_STRING_PERFORMANCE_MODES)
+    result = []
+    if string is not None and string != '':
+        for perf in string.split(';'):
+            perf_dict = {}
+            for line in perf.split(','):
+                key_value = line.split('=')
```

Comment:
If `line.split('=')` always returns list of size == 2, it can be re-written without magic indices `[0]` and `[1]`:
```python
[key, value] = line.split('=')[:2] # [:2] is needed if size can be >= 3
```

Reply:
Nice, sure I'll change this! :+1: